checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -119,6 +120,17 @@ public Void visitBinary(BinaryTree node, Void p) {
         }
         break;
 
+      case PLUS:
+        if (TreeUtils.isStringConcatenation(node)) {
+          if (leftOpType.hasAnnotation(Unsigned.class)) {
```

Comment:
This isn't right because it allows the following:

```
 @UnknownSignedness int unknown = ...;
String s = "" +unknown; // this should be an error.
```

However the following should not be an error:

```
 @UnknownSignedness Date unknown = ...;
String s = "" +unknown; // this should be ok.
```